src/messages.rs

```diff
@@ -261,7 +261,10 @@ impl SignedMessage {
                 .cloned()
                 .collect_vec();
             for irrelevant_id in irrelevant_ids {
-                let _ = self.signatures.remove(&irrelevant_id);
+                let _removed_sig = self.signatures
+                    .remove(&irrelevant_id)
+                    .ok_or(debug!("Tried to remove a signature we did not have : {:?}",
+                                  irrelevant_id));
```

Comment:
I think that would just always print this `debug!`: It's not in a closure, it just gets executed and the resulting `()` then gets passed into `ok_or`. (Same below.)